cmd/phpcs-server/main.go

```diff
@@ -0,0 +1,324 @@
+			if proc.Kind() != "tide" {
+				procs = append(procs, proc)
+			} else {
+				procs = append(procs, &tide.Processor{OutputFile: *flagOutput})
+			}
+		}
+
+		processes = procs
+	}
+
+	// Prepare the Tide Client.
+	err := TideClient.Authenticate(tideConfig.id, tideConfig.secret, tideConfig.authEndpoint)
+	if err != nil {
+		log.Println("Tide API Error:", err)
+		terminateChannel <- struct{}{}
+	}
+
+	// Create a channel that receives messages from a queue, if it does not yet exist.
+	if cMessage == nil {
+		cMessage = messageChannel(messageProvider, buffer)
+	}
+
+	// Poll the message channel until the program is forcefully exited.
+	for {
+		select {
+		// Terminate signal received.
+		case <-terminateChannel:
+			break;
+			// Message received from the queue.
+		case msg := <-cMessage:
+			// If its a single process then don't run as goroutine.
+			if *flagUrl != "" {
+				// We still need to prime the bugger for processMessage() to be able to exit.
+				buffer <- struct{}{}
+				processMessage(msg, TideClient, buffer)
+				os.Exit(0)
+			} else {
+				// Process the message in a go routine.
+				go processMessage(msg, TideClient, buffer)
+			}
+		}
+	}
+}
+
+// messageChannel returns a channel of messages to be processed. The message provider gets polled for
+// the next message and upon success it gets added to the channel.
+func messageChannel(provider message.MessageProvider, buffer chan struct{}) chan *message.Message {
+
+	// Create the message channel.
+	c := make(chan *message.Message)
+
+	// Run this concurrently.
+	go func(b chan struct{}) {
+		for {
+
+			// Get message from provider.
+			msg, err := provider.GetNextMessage()
+
+			// Handle provider errors.
+			if err != nil {
+				// If its a Provider error we might need to panic and fail.
+				if pErr, ok := err.(*message.ProviderError); ok {
+					switch pErr.Type {
+					case message.ErrCritcal:
+						log.Println(pErr)
+						terminateChannel <- struct{}{}
+						break
+					case message.ErrOverQuota:
+						log.Println(pErr, "delaying for 60 seconds")
+						time.Sleep(time.Second * time.Duration(60))
+					}
+				}
+			}
+
+			log.Println("Polling...")
+
+			// If message has been retrieved add it to the channel.
+			if msg != nil {
+				// Block if the buffer is full.
+				b <- struct{}{}
+
+				// Send message to channel.
+				c <- msg
+			}
+			time.Sleep(time.Second * time.Duration(2))
+		}
+	}(buffer)
+
+	// Return the message channel.
+	return c
+}
+
+// processMessage takes an SQS message and runs it through an audit process.
+func processMessage(msg *message.Message, client tideApi.ClientInterface, buffer <-chan struct{}) {
+
+	tidelog.Log(msg.Title, "Processing...")
+
+	var errors []error
+
+	// Initialise result with:
+	// - Tide client reference,
+	// - Temp folder for downloaded/extracted files,
+	// - Audits to run
+	result := &audit.Result{
+		"client":     &client,
+		"tempFolder": tempFolder,
+		"audits": []string{
+			"phpcs_wordpress",
```

Comment:
Same here, we are only allowing two types of audits. Is this not flexible anymore?